test/test.js

```diff
@@ -458,6 +460,27 @@ function checkRefTestResults(browser, id, results) {
   });
 }
 
+function browserTestReportHandler(req, res) {
+  var parsedUrl = url.parse(req.url, true);
+  var pathname = parsedUrl.pathname;
+  if (pathname === '/browserTestReports') {
+    // let information = req.body.browserTestInfo;
+    // var i = 0;
+    // ++i;
+    const writableStream = fs.createWriteStream(
+                           '../coverage/lcov-report/browserData/');
+    // const readable = getReadableStreamSomehow();
+    /* readable.on('data', function(dataToWrite) {
+      writeableStream.write(dataToWrite);
+    })
+
+    readable.on('end',() => {
+      writeableStream.end(null);
+    }) */
+    req.pipe(writableStream);
```

Comment:
This function will take a POST request body and write it to a file.

What you need to do before this is done is:

- Register the `browserTestReportHandler`, similarly as https://github.com/mozilla/pdf.js/blob/a3b2456c7a33811010961ee111c58554dbc74ffc/test/test.js#L218
- Use the [`finish` event of the writableStream](https://nodejs.org/api/stream.html#stream_event_finish) to know when the file has been written, and call the [`end` method of `res`](https://nodejs.org/api/http.html#http_response_end_data_encoding_callback) so that the response ends (and the call to `XMLHttpRequest` finishes).

Reply:
Initially I thought I'll have to define a whole new block of code, something similar to [this](https://github.com/mozilla/pdf.js/blob/master/test/test.js#L542-L568).

However, I guess just adding a line of server.hooks after 218 will do the needful. Also, finalize is already handling all the logging of errors and stuff after that.